packages/cloudvision-connector/test/Wrpc.spec.ts

```diff
@@ -36,16 +38,49 @@ import {
   SUBSCRIBE,
 } from '../src/constants';
 import { log } from '../src/logger';
-import Parser from '../src/parser';
 import { makeToken } from '../src/utils';
-import WRPC from '../src/wrpc';
-import { NotifCallback, SubscriptionIdentifier, RequestArgs } from '../types';
-import { WsCommand, CloudVisionParams } from '../types/params';
-import { CloudVisionQueryMessage } from '../types/query';
+import {
+  CloudVisionParams,
+  CloudVisionQueryMessage,
+  CloudVisionStatus,
+  NotifCallback,
+  QueryParams,
+  RequestArgs,
+  StreamCommand,
+  SubscriptionIdentifier,
+  WsCommand,
+} from '../types';
+
+interface PolymorphicCommandFunction {
+  (command: WsCommand, params: CloudVisionParams, callback: NotifCallback): string;
+}
+
+interface CommandFunction {
+  (p: CloudVisionParams, cb: NotifCallback): string;
+}
+
+interface PostedMessage {
+  source: string;
+  timestamp: number;
+
+  request?: CloudVisionQueryMessage;
+  response?:
+    | {
+        result: { dataset: string };
+        token: string;
+      }
+    | {
+        error: string;
+        status: CloudVisionStatus;
+        token: string;
+      };
+}
+
+type WrpcMethod = 'get' | 'pause' | 'publish' | 'requestService' | 'resume' | 'search';
```

Comment:
reference the string constants?

Reply:
This type refers to the methods of the `Wrpc` class.
"get" and "pause" etc happen to coincide with the method names, but other such as "requestService" do not.